### DISCUSSION TITLE
Oops, something went wrong! QuotaExceededError: Failed to execute 'setItem' on 'Storage': Setting the value of 'chat-next-web-store' exceeded the quota.

### DISCUSSION THREAD

**Discussion by johnrisby** [ChatGPTNextWeb/NextChat#2425] [General]
Just started today after updating to the latest version.

What options do I have other than "Clear all data" which I presume will lose all my previous chats? (I'd rather not).

Thanks

**Comment by reece00** [comment]
You can only clear the data because the upper limit of the browser storage data is reached

**Reply by johnrisby** [reply]
Thanks. I did it earlier anyway as I needed to work. Not a great solution though. Is this something that will always happen after a certain time? Is there nothing in the app to clean up the data without nuking it? Thanks.

**Reply by reece00** [reply]
You can delete a single dialogue to release the storage space. Most people do not encounter this problem because they only have a short chat and will not reach the upper limit of 5MB. You can also replace the Firefox browser because it can set the size of the localStorage

**Reply by johnrisby** [reply]
Thanks, but I use Arc and don't plan to switch to another browser. I've been using this for a few months now and use it quite a lot. I couldn't delete a single dialogue from the browser because it wouldn't display anything due to the lack of error handling when the storage is full (maybe I could have done it manually in developer tools but ...). Be great if something could be added to make this work better. Appreciate most people won't use it as much as I do but eventually everyone actively using it will have this problem over time. I just got there sooner than most :)

**Reply by Yidadaa** [reply]
@johnrisby I will use IndexedDB instead of localStorage to solve this problem.

**Reply by johnrisby** [reply]
@Yidadaa that sounds like a perfect solution, thanks
